src/main/scala/mesosphere/marathon/storage/migration/MigrationTo15.scala

```diff
@@ -138,9 +140,14 @@ private[migration] object MigrationTo15 {
   def deleteEventSubscribers[K, C, S](store: PersistenceStore[K, C, S]): Future[Done] = {
     store match {
       case zk: ZkPersistenceStore =>
-        zk.client.delete("/event-subscribers").map(_ => Done)(ExecutionContexts.callerThread)
-      case _ =>
-        Future.successful(Done)
+        implicit val ec: ExecutionContext = ExecutionContexts.callerThread
+        zk.client.delete("/event-subscribers").map(_ => Done).recover {
+          case _: KeeperException.NoNodeException => Done
```

Comment:
I understand the fix but I don't seem to understand how is that connected to the `disable_store_cache`. Can you clarify?

Reply:
If caching was enabled it was doing nothing, and when it was disabled, if there was no `/event-subscribers`, it crashed.